matchzoo/datapack.py

```diff
@@ -53,18 +53,36 @@ def __init__(self,
         :param left: Store the content or features for id_left.
         :param right: Store the content or features for
             id_right.
-        :param context: Hyper-parameter fitted during
-            pre-processing stage.
         """
         self._relation = relation
         self._left = left
         self._right = right
-        self._context = context
+
+    @property
+    def stage(self):
+        if 'label' in self._relation.columns:
+            return 'train'
+        else:
+            return 'predict'
 
     def __len__(self) -> int:
         """Get numer of rows in the class:`DataPack` object."""
         return self._relation.shape[0]
 
+    def __getitem__(self, index):
```

Comment:
index -> indices?

Reply:
It can be either singular or plural, so use singular to represent both is better.